src/utils/actions.js

```diff
@@ -193,29 +194,31 @@ const baseIgActions = {
     context.flags['valid'] = validation.valid;
   },
   'submit-report': async context => {
-    await submitBundle(context.reportingBundle, context.client.dest)
-      .then(
-        result => {
-          if (result.status === StatusCodes.ACCEPTED || result.status === StatusCodes.OK) {
-            context.flags['submitted'] = true;
-            debug(`/Bundle/${context.reportingBundle.id} submitted to ${context.client.dest}`);
-
-            if (result.data?.resourceType === 'Bundle') {
-              forwardMessageResponse(result.data).then(() =>
-                debug(`Response to /Bundle/${context.reportingBundle.id} forwarded to EHR`)
-              );
+    const listOfPromises = [];
+    const bundleId = context.reportingBundle.id;
+    const urls = getTrustedThirdParties() ?? [context.client.dest];
+    urls.forEach(url => {
+      listOfPromises.push(
+        submitBundle(context.reportingBundle, url)
+          .then(result => {
+            if (result.status === StatusCodes.ACCEPTED || result.status === StatusCodes.OK) {
+              context.flags['submitted'] = true;
+              debug(`/Bundle/${bundleId} submitted to ${url}`);
+
+              if (result.data?.resourceType === 'Bundle') {
+                forwardMessageResponse(result.data).then(() =>
+                  debug(`Response to /Bundle/${bundleId} forwarded to EHR`)
+                );
+              }
             }
-          }
-        },
-        () => {
-          context.flags['submitted'] = false;
-        }
-      )
-      .catch(err => {
-        const bundleId = context.reportingBundle.id;
-        error(`Error submitting Bundle/${bundleId} to ${context.client.dest}\n${err.message}`);
-        context.flags['submitted'] = false;
-      });
+          })
+          .catch(err => {
+            context.flags['submitted'] = false;
```

Comment:
By "last request" do you mean the `forwardMessageResponse`? Even if that fails it would still have been submitted to the PHA/TTP. I don't think we really need another flag for that, but adding a catch with an error message would be helpful for tracking purposes

Reply:
I mean that that the `urls.forEach` will be looping through several urls, but the flag for `submitted` is set to `true` or `false` on every iteration, so the `submitted` flag will just be representative of the latest request to be made.  If the first url fails, but the second succeeds, the flag will be `true`.

Reply:
Good catch, updated to only set once at the end